index.js

```diff
@@ -209,38 +228,40 @@ function updateDesigns(designPaths, callback) {
             });
         });
     }
-    if (!(Array.isArray(designPaths) && typeof callback === 'function')) {
+    if (!(Array.isArray(designPaths))) {
         var msg = 'Invalid parameters';
         log.error(msg);
         throw new VError(msg);
     }
     // everything looks good, we're ready to update any designs that need it
-    var updateDesignFns = [];
+    var updateDesignPromises = [];
     designPaths.forEach(function (designPath) {
         var design = _.get(designs, designPath);
         var parts = designPath.split('.');
         if (!design) {
             log.warn('skipping unknown design ' + designPath);
         } else {
-            updateDesignFns.push(updateDesign.bind(this, parts[0] + ':' + parts[1], parts[2], design));
+            updateDesignPromises.push(updateDesign(parts[0] + ':' + parts[1], parts[2], design));
         }
     });
-    if (updateDesignFns.length === 0) {
-        callback(null, 'no design paths match a design found in couchdb directory');
+    if (updateDesignPromises.length === 0) {
+        return Q('no design paths match a design found in couchdb directory');
     }
-    async.parallel(updateDesignFns, function (error, okMessages) {
-        callback(error ? error : null, okMessages.length ? okMessages : null);
-    });
+    return Q.allSettled(updateDesignPromises);
 }
 
-function updateDesign(dbName, designName, designDoc, callback) {
+function updateDesign(dbName, designName, designDoc) {
     var nanoDb = get(dbName);
+    var deferred = Q.defer();
```

Comment:
for readability: create the deferred promise after the error verification steps
and in those verification steps, simply do a `return Q.reject(new VError(...))`